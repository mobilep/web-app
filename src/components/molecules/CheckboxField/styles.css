.CheckboxField {
	display: flex;
	align-items: center;
	position: relative;
}

.CheckboxField input {
	position: absolute;
	opacity: 0;
	cursor: pointer;
	height: 22px;
	width: 22px;
}

.CheckboxField-checkmark {
	position: relative;
	height: 22px;
	width: 22px;
	border-radius: 50%;
	color: transparent;
	border: 1px solid var(--grey9);
	margin-right: 20px;
	flex-shrink: 0;
}

.CheckboxField input:checked ~ .CheckboxField-checkmark {
	background-color: var(--tomatoRed);
	color: white;
	border: none;
}
