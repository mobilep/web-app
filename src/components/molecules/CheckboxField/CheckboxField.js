import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { CheckIcon } from '../../../icons';

import './styles.css';

const CHECKBOX_POSITION = {
	start: 'start',
	end: 'end',
};

export default class CheckboxField extends Component {

	static propTypes = {
		checked: PropTypes.bool,
		children: PropTypes.any,
		className: PropTypes.string,
		contentPosition: PropTypes.oneOf([CHECKBOX_POSITION.start, CHECKBOX_POSITION.end]),
		name: PropTypes.string,
		onInputChange: PropTypes.func,
	}

	static defaultProps = {
		contentPosition: CHECKBOX_POSITION.end,
	}

	getClassName = (className) => {
		return classnames({
			CheckboxField: true,
		}, className);
	}

	render () {
		const { children, className, name, onInputChange, checked, contentPosition } = this.props;

		return (
			<label className={this.getClassName(className)}>
				{contentPosition === CHECKBOX_POSITION.start && children}
				<Fragment>
					<input type='checkbox'
						name={name}
						onChange={onInputChange}
						checked={checked}
					/>
					<span className='CheckboxField-checkmark'>
						<CheckIcon />
					</span>
				</Fragment>
				{contentPosition === CHECKBOX_POSITION.end && children}
			</label>
		);
	}

}
