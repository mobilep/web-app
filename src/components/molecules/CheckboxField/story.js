import React from 'react';
import { storiesOf } from '@storybook/react';
import CheckboxField from './CheckboxField';

storiesOf('2. Molecules / CheckboxField', module)
	.add('default', () =>
		(<CheckboxField>
			<h1>Hello world</h1>
		</CheckboxField>));
