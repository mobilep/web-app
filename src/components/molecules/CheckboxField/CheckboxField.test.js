import React from 'react';
import { shallow } from 'enzyme';
import CheckboxField from './CheckboxField';

describe('Molecule - CheckboxField', () => {
	it('Should render CheckboxField component', () => {
		const wrapper = shallow(<CheckboxField />);
		expect(wrapper).toMatchSnapshot();
	});
});
