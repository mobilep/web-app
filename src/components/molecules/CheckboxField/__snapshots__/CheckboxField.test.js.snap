// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - CheckboxField Should render CheckboxField component 1`] = `
<label
  className="CheckboxField"
>
  <input
    type="checkbox"
  />
  <span
    className="CheckboxField-checkmark"
  >
    <CheckIcon />
  </span>
</label>
`;
